src/app/features/activity-list/components/submitted-transactions.utils.ts

```diff
@@ -0,0 +1,64 @@
+import { PayloadType, StacksTransaction } from '@stacks/transactions';
+
+import { Tx } from '@app/common/api/transactions';
+import { stacksValue } from '@app/common/stacks-utils';
+import { getTxCaption, getTxSenderAddress } from '@app/common/transactions/transaction-utils';
+
+export function getSubmittedTxCaption(transaction: StacksTransaction) {
+  switch (transaction.payload.payloadType) {
+    case PayloadType.TokenTransfer:
+      return getTxCaption({
+        tx_type: 'token_transfer',
+        tx_id: transaction.txid(),
+      } as Tx);
+    case PayloadType.ContractCall:
+      return getTxCaption({
+        tx_type: 'contract_call',
+        contract_call: {
+          contract_id: `.${transaction.payload.contractName.content}`,
+        },
+      } as unknown as Tx);
+    case PayloadType.SmartContract:
+      return getTxCaption({
+        tx_type: 'smart_contract',
+        smart_contract: {
+          contract_id: `${getTxSenderAddress(transaction)}.${
+            transaction.payload.contractName.content
+          }`,
+        },
+      } as unknown as Tx);
+    default:
+      return null;
+  }
+}
+
+export function getSubmittedTxTitle(transaction: StacksTransaction) {
+  switch (transaction.payload.payloadType) {
+    case PayloadType.TokenTransfer:
+      return 'Stacks Token';
+    case PayloadType.ContractCall:
+      return transaction.payload.functionName.content;
+    case PayloadType.SmartContract:
+      return transaction.payload.contractName.content;
+    default:
+      return '';
+  }
+}
+
+export function getSubmittedTxValue(transaction: StacksTransaction) {
+  switch (transaction.payload.payloadType) {
+    case PayloadType.TokenTransfer:
+      return `-${stacksValue({
+        value: Number(transaction.payload.amount),
+        withTicker: false,
+      })}`;
+    case PayloadType.ContractCall:
+    // TODO: Implement?
+    // return transaction.payload.functionName.content === 'transfer'
+    //   ? cvToValue(transaction.payload.functionArgs[0])
+    //   : null;
```

Comment:
would avoid and keep simple (any method could be called transfer and doesn't necessarily mean transferring assets

Reply:
Ok, yeah, this comment was here so didn't want to just remove w/out checking. I'll remove it.